pkg/vadmin/revive_db_vc.go

```diff
@@ -19,13 +19,49 @@ import (
 	"context"
 	"fmt"
 
+	vops "github.com/vertica/vcluster/vclusterops"
+	"github.com/vertica/vcluster/vclusterops/vstruct"
+	"github.com/vertica/vertica-kubernetes/pkg/net"
 	"github.com/vertica/vertica-kubernetes/pkg/vadmin/opts/revivedb"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 // ReviveDB will initialized a database using an existing communal path. It does
 // this using the vclusterops library.
 func (v *VClusterOps) ReviveDB(ctx context.Context, opts ...revivedb.Option) (ctrl.Result, error) {
-	v.Log.Info("Starting vcluster ReviveDB")
-	return ctrl.Result{}, fmt.Errorf("not implemented")
+	v.Log.Info("Starting VCluster ReviveDB")
+
+	certs, err := v.retrieveHTTPSCerts(ctx)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	s := revivedb.Parms{}
+	s.Make(opts...)
+
+	vcOpts := v.genReviveDBOptions(&s, certs)
+	err = v.VReviveDatabase(vcOpts)
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to revive the database %s: %w", *vcOpts.Name, err)
```

Comment:
We should use `v.logFailure` like we do for create db. This handles updates to events for certain kinds of errors.